src/Microsoft.DotNet.Cli.Utils/ForwardingAppImplementation.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using Microsoft.DotNet.Cli.Utils;
+
+namespace Microsoft.DotNet.Cli.Utils
+{
+    /// <summary>
+    /// A class which encapsulates logic needed to forward arguments from the current process to another process
+    /// invoked with the dotnet.exe host.
+    /// </summary>
+    internal class ForwardingAppImplementation
+    {
+        private const string s_hostExe = "dotnet";
+
+        private readonly string _forwardApplicationPath;
+        private readonly IEnumerable<string> _argsToForward;
+        private readonly string _depsFile;
+        private readonly string _runtimeConfig;
+        private readonly string _additionalProbingPath;
```

Comment:
This is what `ForwardingApp` did previously, I just moved the logic around and wrapped it.